api/v1beta1/azuremanagedcontrolplane_default_test.go

```diff
@@ -53,3 +54,86 @@ func hardcodedAzureManagedControlPlaneWithSSHKey(sshPublicKey string) *AzureMana
 		},
 	}
 }
+
+func TestSetDefaultAutoScalerProfile(t *testing.T) {
+	t.Run("AllFieldsAreNil", func(t *testing.T) {
```

Comment:
I don't think it's entirely necessary to use these `t.Run()` calls and instead just use assertions. You would have to rename the `AzureManagedControlPlane` with different names. 

See this test for examples: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/exp/api/v1beta1/azuremachinepool_default_test.go#L50